Repository Generator - Not Match dataSourceClassName

## Description / Steps to reproduce / Feature proposal

Repository Generator inject (in my case) mySqlDataSource, but should be "**M**ysqlDataSource".

```json
// src/datasources/mysql.datasource.json
{
  "name": "mysql",
  "connector": "mysql",
  "url": "",
  "host": "0.0.0.0",
  "port": 3306,
  "user": "root",
  "password": "",
  "database": "test"
}
```
```javascript
import {DefaultCrudRepository, juggler} from '@loopback/repository';
import {Post} from '../models';
import {mysqlDataSource} from '../datasources';
import {inject} from '@loopback/core';

export class PostRepository extends DefaultCrudRepository<
  Post,
  typeof Post.prototype.id
> {
  constructor(
    @inject('datasources.mysql') protected datasource: mysqlDataSource,
  ) {
    super(Post, datasource);
  }
}
```

```bash
src/repositories/post.repository.ts:3:9 - error TS2724: Module '"/Users/elias/projets/code/loopback/alba/src/datasources/index"' has no exported member 'mysqlDataSource'. Did you mean 'MysqlDataSource'?

3 import {mysqlDataSource} from '../datasources';
          ~~~~~~~~~~~~~~~

  src/datasources/mysql.datasource.ts:5:14
    5 export class MysqlDataSource extends juggler.DataSource {
                   ~~~~~~~~~~~~~~~
    'MysqlDataSource' is declared here.

--- a/packages/cli/generators/repository/index.js
+++ b/packages/cli/generators/repository/index.js
@@ -86,7 +86,7 @@ module.exports = class RepositoryGenerator extends ArtifactGenerator {
     );
 
     this.artifactInfo.dataSourceClassName =
-      this.artifactInfo.dataSourceName + 'DataSource';
+      utils.toClassName(this.artifactInfo.dataSourceName) + 'DataSource';
   }
 
   /**

--- a/packages/cli/generators/service/index.js
+++ b/packages/cli/generators/service/index.js
@@ -205,7 +205,7 @@ module.exports = class ServiceGenerator extends ArtifactGenerator {
     );
 
     this.artifactInfo.dataSourceClassName =
-      this.artifactInfo.dataSourceName + 'DataSource';
+      utils.toClassName(this.artifactInfo.dataSourceName) + 'DataSource';
 
     // Setting up data for templates
     this.artifactInfo.className = utils.toClassName(this.artifactInfo.name);

--- a/packages/cli/test/integration/generators/service.integration.js
+++ b/packages/cli/test/integration/generators/service.integration.js
@@ -86,7 +86,7 @@ describe('lb4 service', () => {
       );
       assert.fileContent(
         expectedFile,
-        /protected datasource: mydsDataSource = new mydsDataSource\(\),/,
+        /protected datasource: mydsDataSource = new MydsDataSource\(\),/,
       );
       assert.fileContent(
         expectedFile,

